### ISSUE TITLE
Testing the new weights reveals issues: they fail to recognize general objects

### ISSUE THREAD

**Issue by Pratik-Ghute** [longzw1997/Open-GroundingDino#24]
"Is this training pipeline fine-tuning pretrained SwinT weights or training from scratch? Testing the new weights reveals issues: they fail to recognize general objects and have incomplete detection on familiar frames.

**Comment by BIGBALLON** [comment]
@Pratik-Ghute pretrain or finetune are OK. Using finetune can save some costs, but if you can obtain more and higher quality data, you can train the model from scratch for higher performance.

**Comment by Pratik-Ghute** [comment]
I wanted to know what this repo is doing finetuning or training from scratch ?

**Comment by BIGBALLON** [comment]
The models we release (1.8M and COCO) are finetuning on GDINO-T 
Some models we not released training from scratch

**Comment by Pratik-Ghute** [comment]
ok and if I train my model using this command:
!python main.py \
        --output_dir "/content/" \
        -c /content/Open-GroundingDino/config/cfg_odvg.py \
        --datasets /content/Open-GroundingDino/config/datasets_od_example.json \
        --pretrain_model_path /content/Open-GroundingDino/weights/groundingdino_swint_ogc.pth \
        --options text_encoder_type=/content/Open-GroundingDino/bert_weights/
        
        It must be finetuning on groundingdino_swint_ogc right?

**Comment by BIGBALLON** [comment]
No, you can use any model, it doesn’t matter fine-tuning or pre-training

**Comment by BIGBALLON** [comment]
No, you can use any model(GDINO pretrain, [DINO](https://github.com/IDEA-Research/DINO) pretrain, [swin](https://github.com/microsoft/Swin-Transformer) pretrain, random init, etc. ), it doesn’t matter fine-tuning or pre-training.

**Comment by Pratik-Ghute** [comment]
ok thanks alot , 

"Requesting clarification on the multiple weight files generated. Is 'checkpoint_best_regular.pth' the weight file for the best epoch with the highest accuracy? 

Additionally, in the 'eval' folder, what are the purposes of 'latest.pth' and '000.pth' weight files?"

**Comment by BIGBALLON** [comment]
@Pratik-Ghute 
1. Is 'checkpoint_best_regular.pth' the weight file for the best epoch with the highest accuracy? yes
2. you can just ignore these files, it dumps some eval metrics for the latest evaluation.

```
>>> x = torch.load("000.pth")
>>> x.keys()
dict_keys(['params', 'counts', 'date', 'precision', 'recall', 'scores'])
```

**Comment by Pratik-Ghute** [comment]
> @Pratik-Ghute pretrain or finetune are OK. Using finetune can save some costs, but if you can obtain more and higher quality data, you can train the model from scratch for higher performance.
> 
> * **For fintune**: choose a reasonable steps(epochs) to get good downstream task performance but avoid overfitting (depends on dataset itself)
> * **For generalization(very important)**: you can use model **ensemble** tech such as [WISE-FT](https://github.com/mlfoundations/wise-ft) to balacne the weights of finetune model and pretrain model.
>   
>   * [OWL-V2](https://github.com/google-research/scenic/tree/main/scenic/projects/owl_vit)  (which is the strongest OVD model now) is also used this model ensemble trick.

will model ensemble tech launch two or more models everytime I inference or take predictions.

**Comment by BIGBALLON** [comment]
as for me, I do [that](https://github.com/mlfoundations/wise-ft) offline and get a new model by ensemble pre-train and fine-tuned model. so nothing changed in inference stage
